src/librustc_codegen_ssa/mir/block.rs

```diff
@@ -21,766 +23,870 @@ use super::place::PlaceRef;
 use super::operand::{OperandValue, OperandRef};
 use super::operand::OperandValue::{Pair, Ref, Immediate};
 
-impl<'a, 'tcx: 'a, Bx: BuilderMethods<'a, 'tcx>> FunctionCx<'a, 'tcx, Bx> {
-    pub fn codegen_block(
-        &mut self,
-        bb: mir::BasicBlock,
-    ) {
-        let mut bx = self.build_block(bb);
-        let data = &self.mir[bb];
+/// Helper structure used by `FunctionCx::codegen_terminator`.
+struct TerminatorCodegenHelper<'a, 'tcx> {
```

Comment:
`TerminatorCodegenHelper` contains the captured values that the old closures `funclet`, `lltarget`, `llblock`, `funclet_br`, and `do_call`.